### PR TITLE
dynet-1201. Fix weight decay load issue.

### PR DESCRIPTION
**Pull request by xunzhang**
Address https://github.com/clab/dynet/issues/1201

### CODE REVIEW

**Comment by xunzhang** [comment]
Please review, @neubig. I tested the performance locally, there is no regression.

File: dynet/tensor.cc [hunk 1]
```diff
@@ -95,6 +96,20 @@ vector<Eigen::DenseIndex> as_vector(const IndexTensor& v) {
...
+  vector<real> res(v.d.size());
+  if (v.device->type == DeviceType::CPU) {
+    memcpy(&res[0], v.v, sizeof(real) * res.size());
+  } else if (v.device->type == DeviceType::GPU) {
+#if HAVE_CUDA
+    CUDA_CHECK(cudaSetDevice(((Device_GPU*)v.device)->cuda_device_id));
+    CUDA_CHECK(cudaMemcpy(&res[0], v.v, sizeof(real) * res.size(), cudaMemcpyDeviceToHost));
+#endif
+  } else { throw std::runtime_error("Bad device type"); }
+  std::transform(res.begin(), res.end(), res.begin(), [&](real t){ return t * a; });
```
Comment (neubig) [inline dynet/tensor.cc]: What about adding `if(a != 1.f)` before this line?

Comment (xunzhang) [reply dynet/tensor.cc]: good catch

**Comment by danielhers** [comment]
@xunzhang assuming I want my code to work with DyNet both before this PR was merged and after, what is the best way to check if weight decay has been applied on model load or if I should apply it myself?

**Pull request merged**